packages/mobile/src/utils/navigation.ts

```diff
@@ -18,3 +22,16 @@ export const getNavigationStateAtRoute = (routeSegments: string[]) => (
     route?.state as NavigationState
   )
 }
+
+/**
+ * Navigation state selector that selects the current route
+ * Can be passed to useNavigationState
+ */
+export const getRoutePath = (state: NavigationState, routePath?: string[]) => {
+  if (!state || state.routes.length === 0) {
+    return routePath
+  }
+
+  const { state: subState, name } = state.routes[state.index]
+  return getRoutePath(subState as NavigationState, [...(routePath ?? []), name])
```

Comment:
nit: is `routePath ?? []` not enough?

Reply:
Wouldn't that result in a nested array instead of a flat array? The output rn is something like `['main', 'feed', 'track']`

Reply:
oh yeah i wasn't looking correctly, yup!